### PR TITLE
Add cli testing and fix improper mismatch debug message

### PR DESCRIPTION
**Pull request by mikemrm**
Adds the following tests
```
packetnetworking/test_cli.py::test_cli[rootfs required]
packetnetworking/test_cli.py::test_cli[rootfs: --rootfs]
packetnetworking/test_cli.py::test_cli[rootfs: -t]
packetnetworking/test_cli.py::test_cli[--metadata-url/file undefined]
packetnetworking/test_cli.py::test_cli[--metadata-url defined]
packetnetworking/test_cli.py::test_cli[--metadata-file defined]
packetnetworking/test_cli.py::test_cli[Operating System: --operating-system]
packetnetworking/test_cli.py::test_cli[Operating System: -o]
packetnetworking/test_cli.py::test_cli[--resolvers defined]
packetnetworking/test_cli.py::test_cli[verbose level 1 (INFO): -v]
packetnetworking/test_cli.py::test_cli[verbose level 2 (DEBUG): -v]
packetnetworking/test_cli.py::test_cli[verbose level 3+ (DEBUG): -v]
packetnetworking/test_cli.py::test_cli[verbose level 1 (INFO): --verbose]
packetnetworking/test_cli.py::test_cli[verbose level 2 (DEBUG): --verbose]
packetnetworking/test_cli.py::test_cli[verbose level 3+ (DEBUG): --verbose]
packetnetworking/test_cli.py::test_cli[quiet long option (--quiet)]
packetnetworking/test_cli.py::test_cli[quiet short option (-q)]
packetnetworking/test_cli.py::test_cli_retries_on_exception[Max Attempts long option (--max-attempts)]
packetnetworking/test_cli.py::test_cli_retries_on_exception[Max Attempts short option (-n)]
packetnetworking/test_cli.py::test_cli_retries_on_exception[Attempt -1: no sleep]
packetnetworking/test_cli.py::test_cli_retries_on_exception[Attempt 0: no sleep]
packetnetworking/test_cli.py::test_cli_retries_on_exception[Attempt 1: no sleep]
packetnetworking/test_cli.py::test_cli_retries_on_exception[Attempt 2: sleep 4 seconds]
packetnetworking/test_cli.py::test_cli_retries_on_exception[Attempt 7: sleep 128 seconds]
packetnetworking/test_cli.py::test_cli_retries_on_exception[Attempt 8: sleep 128 seconds]
packetnetworking/test_cli.py::test_try_run_with_url
packetnetworking/test_cli.py::test_try_run_with_file
packetnetworking/test_cli.py::test_setup_builder[URL]
packetnetworking/test_cli.py::test_setup_builder[File]
packetnetworking/test_cli.py::test_setup_builder[File+URL]
packetnetworking/test_cli.py::test_set_os[None]
packetnetworking/test_cli.py::test_set_os[Empty]
packetnetworking/test_cli.py::test_set_os[No Space]
packetnetworking/test_cli.py::test_set_os[Too Many Spaces]
packetnetworking/test_cli.py::test_set_os[Matches]
packetnetworking/test_cli.py::test_set_os[Diff: Missing]
packetnetworking/test_cli.py::test_set_os[Diff: Distro Case]
packetnetworking/test_cli.py::test_set_os[Diff: Version Case]
packetnetworking/test_cli.py::test_set_resolvers[None (With metadata)]
packetnetworking/test_cli.py::test_set_resolvers[Empty (With metadata)]
packetnetworking/test_cli.py::test_set_resolvers[None (Without metadata]
packetnetworking/test_cli.py::test_set_resolvers[Empty (Without metadata]
packetnetworking/test_cli.py::test_set_resolvers[Single Resolver]
packetnetworking/test_cli.py::test_set_resolvers[Multiple Resolvers]
packetnetworking/test_cli.py::test_set_resolvers[Multiple Resolvers with extra commas]
```

### CODE REVIEW

**Comment by dustinmiller1337** [comment]
Looks great 👍

File: packetnetworking/test_cli.py [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+    assert mocked_log.error.call_count == 1
+    assert mocked_time_sleep.call_count == 1
+    mocked_time_sleep.assert_called_with(4)
+
+    # Exception raised, therefore bad exit code
+    assert result.exit_code != 0
+
+
+def test_try_run_with_url(mockit, metadata):
+    with patch("requests.get") as mocked_requests_get:
```
Comment (mmlb) [inline packetnetworking/test_cli.py:376]: these mocks can all be in just one `with` statement if separated with `,`. Might get very long, but that might be better than all the extra indentation :D.

**Comment by mikemrm** [comment]
> Could do with some `black` right?

Black was already used or tests wouldn't have passed.

File: packetnetworking/test_cli.py (hunk 1 above)
Comment (mikemrm) [reply packetnetworking/test_cli.py:376]: interesting, will look into that.

File: packetnetworking/cli.py [hunk 2]
```diff
@@ -137,7 +158,7 @@ def try_run(
             os.orig_distro = os.distro
             os.orig_version = os.version
             os.distro = os_name.lower()
-            os.version = os_version
+            os.version = os_version.lower()
```
[the remark this answers is not in the document]
Comment (mikemrm) [reply packetnetworking/cli.py:169]: yep, you're right. will patch that

**Comment by mmlb** [comment]
> > Could do with some `black` right?
> 
> Black was already used or tests wouldn't have passed.

Ahh ok, I ran black from latest nixpkgs I had and it changed some stuff so I just assumed it wasn't use, my bad.

**Pull request merged**